api/src/main/java/org/pmiops/workbench/genomics/GenomicExtractionService.java

```diff
@@ -274,4 +276,26 @@ public GenomicExtractionJob submitGenomicExtractionJob(DbWorkspace workspace, Db
 
     return new GenomicExtractionJob().status(TerraJobStatus.RUNNING);
   }
+
+  public void abortGenomicExtractionJob(DbWorkspace dbWorkspace, String jobId) throws ApiException {
+    Optional<DbWgsExtractCromwellSubmission> dbSubmission =
+        wgsExtractCromwellSubmissionDao.findByWorkspaceWorkspaceIdAndWgsExtractCromwellSubmissionId(
+            dbWorkspace.getWorkspaceId(), Long.valueOf(jobId));
+
+    if (dbSubmission.isPresent()) {
+      WgsCohortExtractionConfig cohortExtractionConfig =
+          workbenchConfigProvider.get().wgsCohortExtraction;
+
+      submissionApiProvider
+          .get()
+          .abortSubmission(
+              cohortExtractionConfig.operationalTerraWorkspaceNamespace,
+              cohortExtractionConfig.operationalTerraWorkspaceName,
+              dbSubmission.get().getSubmissionId());
+
+    } else {
+      throw new BadRequestException(
```

Comment:
nit: `NotFoundException` I think is more appropriate here, since from the API perspective the full resource name includes the workspace and dataset IDs